src/argv.c

```diff
@@ -315,6 +315,29 @@ format_expand_arg(struct format_context *format, const char *name, const char *e
 			return false;
 		return string_format_from(format->buf, &format->bufpos, "%s", value);
 	}
+	if (!prefixcmp(name, "%(sh")) {
+		char cbuf[SIZEOF_STR];
+		const char *c="";
+		char value[SIZEOF_STR]={0};
+		const char *cstart = name + STRING_SIZE("%(sh");
+		const int clen = end - cstart - 1;
+		int size;
+		FILE *cp;
+		if (end && clen > 0 && string_format(cbuf, "sh -c '%.*s'", clen, cstart)) {
```

Comment:
if the command can contain single quotes, this will go wrong.
We should either escape single quotes (replacing them with `'\''`) or,
better, don't spawn two shells here, one is enough.

Reply:
Yes, I think  that the subshell can be skipped, it would be concise with the usual ``bind generic {key} !sh -c '…'`` wrapping ↔ no wrapping → no fancy features like env-vars.